FEXCore/Source/Interface/IR/IR.json

```diff
@@ -316,42 +324,22 @@
         "HasSideEffects": true
       },
 
-      "GPRPair = CPUID GPR:$Function, GPR:$Leaf": {
+      "CPUID GPR:$Function, GPR:$Leaf, GPR:$OutEAX, GPR:$OutEBX, GPR:$OutECX, GPR:$OutEDX": {
```

Comment:
The main thing I want in this PR is a way to read our IR.json and know which arguments are input, output, or rmw. Even if it is purely a textual representation that nothing parses today.

Maybe something in each IR definition that uses this new output, or rmw members like:
```json
"ArgumentTypes": {
  "OutEAX": "Output",
  "OutEBX": "Output",
  "OutECX": "Output",
  "OutEDX": "Output",
},
```

Then in the case of like Pop where an input is RMW
```json
"ArgumentTypes": {
  "InoutAddr": "RMW",
},
```

And something like CASPair where there is a mixture:
GPR:$ExpectedLo, GPR:$ExpectedHi, GPR:$DesiredLo, GPR:$DesiredHi, GPR:$Addr, GPR:$OutLo, GPR:$OutHi
```json
"ArgumentTypes": {
  "ExpectedLo": "Input",
  "ExpectedHi": "Input",
  "DesiredLo": "Input",
  "DesiredHi": "Input",
  "Addr": "Input",
  "OutLo": "Output",
  "OutHigh": "Output",
},
```

Even though we don't parse any of this, I want to be able to come to the JSON in a few months time and not have to wrack my brain trying to remember which SSA arguments are input, output, and RMW, especially if the name of the argument doesn't always imply which one.

Then maybe in the future we can also add some validation to this, which will make reparsing this less of a nightmare as well.

Reply:
That`s very fair, yeah! I will noodle on this. I don`t love that syntax but will do it if I can`t come up with anything better.

Reply:
Reworked.